projects/client/RabbitMQ.Client/src/client/impl/SocketFrameHandler.cs

```diff
@@ -1,321 +1,304 @@
-        {
-            set
-            {
-                try
-                {
-                    if (m_socket.Connected)
-                    {
-                        m_socket.ReceiveTimeout = value;
-                    }
-                }
-                catch (SocketException)
-                {
-                    // means that the socket is already closed
-                }
-            }
-        }
-
-        public int WriteTimeout
-        {
-            set
-            {
-                m_writeableStateTimeout = value;
-                m_socket.Client.SendTimeout = value;
-            }
-        }
-
-        public void Close()
-        {
-            lock (_semaphore)
-            {
-                if (!_closed)
-                {
-                    try
-                    {
-                        m_socket.Close();
-                    }
-                    catch (Exception)
-                    {
-                        // ignore, we are closing anyway
-                    }
-                    finally
-                    {
-                        _closed = true;
-                    }
-                }
-            }
-        }
-
-        public Frame ReadFrame()
-        {
-            lock (m_reader)
-            {
-                return Frame.ReadFrom(m_reader);
-            }
-        }
-
-        public void SendHeader()
-        {
-            lock (m_writer)
-            {
-                m_writer.Write(Encoding.ASCII.GetBytes("AMQP"));
-                if (Endpoint.Protocol.Revision != 0)
-                {
-                    m_writer.Write((byte)0);
-                    m_writer.Write((byte)Endpoint.Protocol.MajorVersion);
-                    m_writer.Write((byte)Endpoint.Protocol.MinorVersion);
-                    m_writer.Write((byte)Endpoint.Protocol.Revision);
-                }
-                else
-                {
-                    m_writer.Write((byte)1);
-                    m_writer.Write((byte)1);
-                    m_writer.Write((byte)Endpoint.Protocol.MajorVersion);
-                    m_writer.Write((byte)Endpoint.Protocol.MinorVersion);
-                }
-                m_writer.Flush();
-            }
-        }
-
-        public void WriteFrame(Frame frame)
-        {
-            lock (m_writer)
-            {
-                m_socket.Client.Poll(m_writeableStateTimeout, SelectMode.SelectWrite);
-                frame.WriteTo(m_writer);
-                m_writer.Flush();
-            }
-        }
-
-        public void WriteFrameSet(IList<Frame> frames)
-        {
-            lock (m_writer)
-            {
-                m_socket.Client.Poll(m_writeableStateTimeout, SelectMode.SelectWrite);
-                foreach(var f in frames)
-                {
-                    f.WriteTo(m_writer);
-                }
-                m_writer.Flush();
-            }
-        }
-
-        public void Flush()
-        {
-            lock (m_writer)
```

Comment:
Removed this lock and flush, since we don't require flush when not using a BufferedStream.